ArduCopter/events.cpp

```diff
@@ -16,8 +16,10 @@ void Copter::failsafe_radio_on_event()
     } else {
         if (control_mode == AUTO && g.failsafe_throttle == FS_THR_ENABLED_CONTINUE_MISSION) {
             // continue mission
-        } else if (control_mode == LAND && g.failsafe_battery_enabled == FS_BATT_LAND && failsafe.battery) {
-            // continue landing
+        } else if (control_mode == LAND &&
+                   battery.has_failsafed() &&
+                   battery.get_highest_failsafe_priority() >= FAILSAFE_LAND_PRIORITY) {
```

Comment:
Good catch. The removal of control mode is a behavior change, and probably not a desired one. Consider a low battery failsafe that starts a landing, but the user takes over in stabilize for example to avoid hitting something. That's perfectly reasonable, but on a radio failsafe we shouldn't stay in stabilize for example. (If we wanted to retrigger land then I can see that being very reasonable, although that is a behavior change I don't want to push at the moment)

Reply:
Ok, I guess that's fine. This is going to be the next issue we'll have: priorities between different failsafes.